### PR TITLE
Document AD principal conflicts

### LINKED ISSUE #406: Document that an AD realm must not be shared between multiple k8s clusters
Followup to #402.

- [ ] Add note to docs about each K8s cluster requiring its own AD realm.
- [ ] Detect conflicts during AD provisioning, and link to the docs above.

### PR DESCRIPTION
**Pull request by nightkr**
# Description

Fixes #406.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [X] Changes are OpenShift compatible
- [X] Helm chart can be installed and deployed operator works
- [X] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: rust/krb5-provision-keytab/src/active_directory.rs [hunk 1]
```diff
@@ -60,7 +60,8 @@ pub enum Error {
     CreateLdapUser { source: ldap3::LdapError },
 
     #[snafu(display(
-        "LDAP user already exists, either delete it manually or add it to the password cache ({password_cache_ref})"
+        "LDAP user already exists but is missing from the password cache ({password_cache_ref}) (hint: see {link})",
+        link = "https://docs.stackable.tech/home/nightly/secret-operator/troubleshooting.html#active-directory-ldap-user-conflict"
```
Comment (nightkr) [inline rust/krb5-provision-keytab/src/active_directory.rs:64]: https://github.com/stackabletech/operator-rs/issues/768

File: docs/modules/secret-operator/pages/troubleshooting.adoc [hunk 2]
```diff
@@ -16,3 +20,23 @@
...
+The Stackable Secret Operator maintains a cache of Active Directory user credentials. This error occurs when a required user is missing from the cache but exists in AD.
+
+This can be caused by a few different root issues:
+
+1. A race condition where multiple Pods require the same identity at the same time, leading to a stale cache.
+   This is transient, and should resolve itself within a few seconds as the cache is updated and any failed attempts are retried.
+2. Trying to reuse a single AD domain across multiple Kubernetes clusters.
+   This takes care to do safely, please see xref:secretclass.adoc#ad-principal-conflicts[Principal Conflicts] for more information.
+3. Deleting a user from the cache but not from AD (including deleting the whole cache).
+   When deleting users from the cache, make sure to also delete the corresponding AD users. Secret Operator should then automatically recreate them.
```
Comment (maltesander) [inline docs/modules/secret-operator/pages/troubleshooting.adoc:39]:
```suggestion
   When deleting users from the cache, make sure to also delete the corresponding AD users. The Secret Operator should then automatically recreate them.
```

**Review by maltesander — DISMISSED** [review]
LGTM!

**Review by maltesander — APPROVED** [review]
LGTM!

**Pull request merged**